### PR TITLE
(docs/1204) - Updated the APIs and documentation for Search Microservice

### PR DESCRIPTION
**Pull request by riginoommen**
# Resolves 1204

# Explain the fix

- Reviewed documentation for Search microservice and updated the APIs
- Renaming the May Jun blog md file(updating date name in the file name)

## Does this PR introduce a breaking change
No

## Screenshots
N/A

### Ready-for-merge Checklist

- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?

### CODE REVIEW

File: packages/documentation-spa/docs/microservices/search/search-service.md [hunk 1]
```diff
@@ -89,11 +89,11 @@ In the GraphQL GET Operations are defined as Queries and POST/PUT/PATCH operatio
 
 ### Search Query
 
-- Operation Name - search
-    - Supported Query Variables - query, start, rows
+- Operation Name - Search
+- Supported Query Variables - ```query```, ```start```, ```rows```
```
Comment (deshmukhmayur) [inline packages/documentation-spa/docs/microservices/search/search-service.md:93]: ` ``` ` is usually for blockquotes, so just single ` is needed.

```suggestion
- Supported Query Variables - `query`, `start`, `rows`
```

File: packages/documentation-spa/docs/microservices/search/search-service.md [hunk 2]
```diff
@@ -89,11 +89,11 @@ In the GraphQL GET Operations are defined as Queries and POST/PUT/PATCH operatio
...
 
 Example Query
-```graphql
+```sh
```
Comment (deshmukhmayur) [inline packages/documentation-spa/docs/microservices/search/search-service.md:96]: This adds proper syntax highlighting in github markdown for graphql queries.
```suggestion
```graphql
```

File: packages/documentation-spa/docs/microservices/search/search-service.md [hunk 3]
```diff
@@ -114,18 +114,57 @@ query Search($query: String, $start: Int, $rows: Int) {
...
 ```
-## Mutations
+### List Search Maps Query
 
-In search microservice every record has a unique key called id which is used to manage the mutation.
+- Operation Name - ListSearchMap
 
-1. Add/Update new index
+Example Query
+```sh
```
Comment (deshmukhmayur) [inline packages/documentation-spa/docs/microservices/search/search-service.md:122]:
```suggestion
```graphql
```

File: packages/documentation-spa/docs/microservices/search/search-service.md [hunk 4]
```diff
@@ -114,18 +114,57 @@ query Search($query: String, $start: Int, $rows: Int) {
...
+```
+### List Search Map by ```_id```
 
-### Operation Name - manageIndex
+- Operation Name - GetSearchMap
+- Supported Query Variables - ```_id_```
 
-- Required Mutation Variables - SearchInput
+Example Query
+```sh
```
Comment (deshmukhmayur) [inline packages/documentation-spa/docs/microservices/search/search-service.md:143]:
```suggestion
```graphql
```

File: packages/documentation-spa/docs/microservices/search/search-service.md [hunk 5]
```diff
@@ -114,18 +114,57 @@ query Search($query: String, $start: Int, $rows: Int) {
...
+
+In search microservice every record has a unique key called id which is used to manage the mutation.
+
+### Create/Update data in index
+- Operation Name - ManageIndex
+- Required Mutation variables type - ```SearchInput```
 
 Example Mutation
-```graphql
+```sh
```
Comment (deshmukhmayur) [inline packages/documentation-spa/docs/microservices/search/search-service.md:167]:
```suggestion
```graphql
```

File: packages/documentation-spa/docs/microservices/search/search-service.md [hunk 6]
```diff
@@ -134,27 +173,88 @@ mutation ManageIndex($input: SearchInput) {
...
+### Delete data from index
 
-    - Required Mutation Variables - name, title, uid, rhatUUID, memberOf, createdBy, createdOn, updatedBy, updatedOn
+- Operation Name - DeleteIndex
+- Required Mutation Variable - ```id```
 
 Example Mutation
 
-```graphql
+```sh
```
Comment (deshmukhmayur) [inline packages/documentation-spa/docs/microservices/search/search-service.md:198]:
```suggestion
```graphql
```

File: packages/documentation-spa/docs/microservices/search/search-service.md [hunk 7]
```diff
@@ -134,27 +173,88 @@ mutation ManageIndex($input: SearchInput) {
...
         status
     }
 }
 ```
+### Create Search Map
+- Operation Name - CreateSearchMap
+- Required Mutation variables type - ```SearchMapInput```
+
+Example Mutation
+```sh
```
Comment (deshmukhmayur) [inline packages/documentation-spa/docs/microservices/search/search-service.md:210]:
```suggestion
```graphql
```

File: packages/documentation-spa/docs/microservices/search/search-service.md [hunk 8]
```diff
@@ -134,27 +173,88 @@ mutation ManageIndex($input: SearchInput) {
...
+    }
+}
+```
+
+### Update Search Map
+- Operation Name - UpdateSearchMap
+- Required Mutation variables type - ```SearchMapInput```
+
+Example Mutation
+```sh
```
Comment (deshmukhmayur) [inline packages/documentation-spa/docs/microservices/search/search-service.md:231]:
```suggestion
```graphql
```

File: packages/documentation-spa/docs/microservices/search/search-service.md [hunk 9]
```diff
@@ -134,27 +173,88 @@ mutation ManageIndex($input: SearchInput) {
...
+```
+
+### Delete Search Map
+
+- Operation Name - DeleteSearchMap
+- Required Mutation Variable - ```id```
+
+Example Mutation
+
+```sh
```
Comment (deshmukhmayur) [inline packages/documentation-spa/docs/microservices/search/search-service.md:254]:
```suggestion
```graphql
```

**Review by deshmukhmayur — CHANGES_REQUESTED** [review]

File: packages/documentation-spa/docs/microservices/search/search-service.md (hunk 1 above)
Comment (riginoommen) [reply packages/documentation-spa/docs/microservices/search/search-service.md]: Changed 👍🏻

File: packages/documentation-spa/docs/microservices/search/search-service.md (hunk 2 above)
Comment (riginoommen) [reply packages/documentation-spa/docs/microservices/search/search-service.md]: Changed 👍🏻

File: packages/documentation-spa/docs/microservices/search/search-service.md (hunk 3 above)
Comment (riginoommen) [reply packages/documentation-spa/docs/microservices/search/search-service.md]: Changed 👍🏻

File: packages/documentation-spa/docs/microservices/search/search-service.md (hunk 4 above)
Comment (riginoommen) [reply packages/documentation-spa/docs/microservices/search/search-service.md]: Changed 👍🏻

File: packages/documentation-spa/docs/microservices/search/search-service.md (hunk 5 above)
Comment (riginoommen) [reply packages/documentation-spa/docs/microservices/search/search-service.md]: Changed 👍🏻

File: packages/documentation-spa/docs/microservices/search/search-service.md (hunk 6 above)
Comment (riginoommen) [reply packages/documentation-spa/docs/microservices/search/search-service.md]: Changed 👍🏻

File: packages/documentation-spa/docs/microservices/search/search-service.md (hunk 7 above)
Comment (riginoommen) [reply packages/documentation-spa/docs/microservices/search/search-service.md]: Changed 👍🏻

File: packages/documentation-spa/docs/microservices/search/search-service.md (hunk 8 above)
Comment (riginoommen) [reply packages/documentation-spa/docs/microservices/search/search-service.md]: Changed 👍🏻

File: packages/documentation-spa/docs/microservices/search/search-service.md (hunk 9 above)
Comment (riginoommen) [reply packages/documentation-spa/docs/microservices/search/search-service.md]: Changed 👍🏻

File: packages/documentation-spa/docs/microservices/search/search-service.md [hunk 10]
```diff
@@ -7,9 +7,8 @@ slug: /microservices/search-service
 
 ## Developers
 
-## Component Contributors
-
 1. Rigin Oommen - [roommen@redhat.com](mailto:roommen@redhat.com) - [riginoommen (Rigin Oommen) · GitHub](https://github.com/riginoommen)
+2. Ghanshyam Lohar - [glohar@redhat.com](mailto:glohar@redhat.com)
```
Comment (ghanlohar) [inline packages/documentation-spa/docs/microservices/search/search-service.md:11]:
```suggestion
2. Ghanshyam Lohar - [glohar@redhat.com](mailto:glohar@redhat.com) - [ghanlohar (Ghanshyam Lohar) · GitHub](https://github.com/ghanlohar)
```

File: packages/documentation-spa/docs/microservices/search/search-service.md [hunk 11]
```diff
@@ -159,3 +192,100 @@ mutation DeleteIndex($id: String!) {
...
+                variables: {
+                    input: mode === 'index' ? data?.input : data,
+                    mode
+                }
+            }),
+        }).then((response: any) => {
+            return response.json();
+        })
+        .then((result: any) => {
+            const statuses = [200, 204];
```
Comment (ghanlohar) [inline packages/documentation-spa/docs/microservices/search/search-service.md:280]:
```suggestion
            const successStatusCodes = [200, 204];
```

File: packages/documentation-spa/docs/microservices/search/search-service.md [hunk 12]
```diff
@@ -159,3 +192,100 @@ mutation DeleteIndex($id: String!) {
...
+            if (statuses.includes(result.data?.manageIndex?.status)) {
```
Comment (ghanlohar) [inline packages/documentation-spa/docs/microservices/search/search-service.md:281]:
```suggestion
            if (successStatusCodes.includes(result.data?.manageIndex?.status)) {
```

File: packages/documentation-spa/docs/microservices/search/search-service.md [hunk 13]
```diff
@@ -159,3 +192,100 @@ mutation DeleteIndex($id: String!) {
...
+                console.info('Sucessfully completed the index updation')
+            } else if (!statuses.includes(result.data?.manageIndex?.status)) {
```
Comment (ghanlohar) [inline packages/documentation-spa/docs/microservices/search/search-service.md:283]:
```suggestion
            } else if (!successStatusCodes.includes(result.data?.manageIndex?.status)) {
```

**Review by ghanlohar — CHANGES_REQUESTED** [review]

File: packages/documentation-spa/docs/microservices/search/search-service.md (hunk 10 above)
Comment (riginoommen) [reply packages/documentation-spa/docs/microservices/search/search-service.md]: Changed 👍🏻

File: packages/documentation-spa/docs/microservices/search/search-service.md (hunk 11 above)
Comment (riginoommen) [reply packages/documentation-spa/docs/microservices/search/search-service.md]: Changed 👍🏻

File: packages/documentation-spa/docs/microservices/search/search-service.md (hunk 12 above)
Comment (riginoommen) [reply packages/documentation-spa/docs/microservices/search/search-service.md]: Changed 👍🏻

File: packages/documentation-spa/docs/microservices/search/search-service.md (hunk 13 above)
Comment (riginoommen) [reply packages/documentation-spa/docs/microservices/search/search-service.md]: Changed 👍🏻

**Review by deshmukhmayur — APPROVED** [review]

File: packages/documentation-spa/docs/microservices/search/search-service.md [hunk 14]
```diff
@@ -115,40 +113,75 @@ query Search($query: String, $start: Int, $rows: Int) {
...
-                title
-                abstract
-                description
-                icon
-                uri
-                tags
-                timestamp
-            }]
-        }
+mutation ManageIndex($input: SearchInput, $mode: String) {
```
Comment (ghanlohar) [inline packages/documentation-spa/docs/microservices/search/search-service.md:163]:
```suggestion
mutation ManageIndex($input: SearchInput, $mode: String!) {
```

File: packages/documentation-spa/docs/microservices/search/search-service.md [hunk 15]
```diff
@@ -159,3 +192,100 @@ mutation DeleteIndex($id: String!) {
...
+}
+```
+## Code Snippets
+#### Universal Helper code for indexing the data with search microservice.
+
+- Language - Javascript
+```js
+manageSearchIndex(data: any, mode: string) {
+    let query: string = `
+    mutation ManageIndex($input: SearchInput, $mode: String) {
```
Comment (ghanlohar) [inline packages/documentation-spa/docs/microservices/search/search-service.md:256]:
```suggestion
    mutation ManageIndex($input: SearchInput, $mode: String!) {
```

**Review by ghanlohar — CHANGES_REQUESTED** [review]

File: packages/documentation-spa/docs/microservices/search/search-service.md (hunk 14 above)
Comment (riginoommen) [reply packages/documentation-spa/docs/microservices/search/search-service.md]: Changed

File: packages/documentation-spa/docs/microservices/search/search-service.md (hunk 15 above)
Comment (riginoommen) [reply packages/documentation-spa/docs/microservices/search/search-service.md]: Changed

**Review by ghanlohar — APPROVED** [review]
LGTM

**Review by ghanlohar — APPROVED** [review]

**Pull request merged**